docs/shared/recipes/module-federation/manage-library-versions.md

```diff
@@ -0,0 +1,129 @@
+
+A challenge arises when these federated modules are integrated into a _host_ or other _remotes_. Given that each federated module carries its own dependencies, the host application may have inadvertently downloaded multiple copies of the same dependency. This redundancy does two things:
+
+1. Multiple copies of the same dependency can create bottlenecks and conflicts, resulting in unexpected behaviour.
+2. With redundant dependencies, the host application can become bloated, increasing the bandwidth and consuming more memory and resources on the user's device.
+
+To mitigate these issues, Module Federation has a shared API. Its primary function is to act as a gatekeeper, ensuring that only one copy of a dependency is downloaded, regardless of how many federated modules request it.
+
+### How it works
+
+The Shared API maintains a registry of all the downloaded dependencies. When a federated module requests a dependency, the Shared API checks the registry. If the dependency already exists, the module is directed to use the existing copy. If not, the dependency is downloaded and added to the registry.
+
+![How Shared API works](/shared/guides/module-federation/shared-api.png)
+
+{% callout type="info" title="Lost?" %}
+If you are not familiar with the concepts of federated modules, remotes, and hosts, please read the [Faster builds with module federation](https://nx.dev/concepts/more-concepts/faster-builds-with-module-federation) for an introduction.
+{% /callout %}
+
+## Our Approach
+
+Although the Shared API is a powerful tool, it can be challenging to manage. The Shared API is configured in the Module Federation Config File, which is a JavaScript or TypeScript file. This file is not part of the build process, so should you want to use a different version of a workspace dependency, you would have to manually record the change outside of the build process which can be a tedious and error-prone.
+
+**Nx** follows the Single-Responsibility Principle (SRP) for managing library versions. Each library is responsible for its version. This approach is beneficial because it allows you to update a library without updating the entire application. Additionally, with this approach, consumers of the library (remotes/hosts) can dictate which version of the library they want to use, which can be further predicated by versioning schemes like SemVer.
+
+## How are library versions managed?
+
+With **Nx** there are tree ways to manage how library versions are shared / managed with Module Federation:
+
+1. Opt in to sharing library versions
+
+This is the default behaviour for **Nx**. All dependencies are **singletons** and will be shared between remotes and hosts.
+
+2. Opt out from sharing library versions
```

Comment:
```suggestion
#### 2. Opt out from sharing library versions
```

Reply:
If you don't turn these into headers, the second bullet point is changed to a `1`